main.py

```diff
@@ -2154,12 +2158,54 @@ def do_command(self, command, args, sender, channel, tags, isWhisper=False):
                                          **msgArgs), isWhisper)
                 return
             if command == "points":
-                # print("Checking points for " + sender)
-                cur = db.cursor()
-                cur.execute("SELECT points FROM users WHERE id = %s", [tags['user-id']])
-                self.message(channel, str(tags['display-name']) + ", you have " + str(cur.fetchone()[0]) + " points!",
-                             isWhisper=isWhisper)
-                cur.close()
+                with db.cursor() as cur:
+                    cur.execute("SELECT points FROM users WHERE id = %s", [tags['user-id']])
+                    points = cur.fetchone()[0]
+                    pudding = sum(getPuddingBalance(tags['user-id']))
+                    self.message(channel, "%s, you have %d points and %d pudding!" % (tags['display-name'], points, pudding), isWhisper)
+                    return
+            if command == "pudding":
+                subcmd = "" if len(args) < 1 else args[0].lower()
+                if subcmd == "booster":
+                    if len(args) < 2:
+                        self.message(channel, "Usage: !pudding booster <name>", isWhisper)
+                        return
+                    # check that the pack is actually buyable
+                    with db.cursor() as cur:
+                        cur.execute("SELECT name, cost FROM boosters WHERE name = %s AND buyable = 1", [args[1]])
+                        booster = cur.fetchone()
+                        if booster is None:
+                            self.message(channel, "Invalid booster specified.", isWhisper)
+                            return
+                        cost = math.ceil(int(booster[1])/int(config["puddingExchangeRate"]))
+                        if not hasPudding(tags['user-id'], cost):
+                            self.message(channel, "%s, you can't afford a %s booster. They cost %d pudding." % (tags['display-name'], booster[0], cost), isWhisper)
+                            return
+                        takePudding(tags['user-id'], cost)
+                        try:
+                            openBooster(self, tags['user-id'], sender, tags['display-name'], channel, isWhisper, booster[0])
+                            if checkHandUpgrade(tags['user-id']):
+                                messageForHandUpgrade(tags['user-id'], tags['display-name'], self, channel, isWhisper)
+                            self.message(channel, "%s, you open a %s booster for %d pudding: %s/booster?user=%s" % (tags['display-name'], booster[0], cost, config["siteHost"], sender), isWhisper)
+                        except InvalidBoosterException:
+                            self.message(channel,
+                                        "Go tell an admin that this booster type is broken.",
```

Comment:
Is there a reason to make the user deal with this? You now have a way to notify admins via Discord.

Telling an admin may an issue for casual players who may not know any admin's name.

Reply:
This is just a generic message (since this error should almost never be thrown barring a huge misconfiguration) but that is a worthwhile point.